libpromises/promises.c

```diff
@@ -240,6 +250,19 @@ Promise *DeRefCopyPromise(EvalContext *ctx, const Promise *pp)
 
                         if (IsDefinedClass(ctx, scp->classes))
                         {
+                            SeqAppend(constraints_to_expand, scp);
+                        }
+                    }
+
+                    SeqReverse(constraints_to_expand);
+                    // Only do the copy expansion if we copied something while inheriting
+                    for (size_t r = 0; r < SeqLength(constraints_to_expand); r++)
```

Comment:
OK; done. I got segfaults because size_t can't be negative and forgot to investigate, thanks for catching that.